### ISSUE TITLE
Add suggested completions for key chords

### ISSUE THREAD

**Issue by pcgeek86** [Microsoft/vscode#3969]
With more complex key chords, it would be incredibly helpful to have "suggested" completions, based on the first element of the chord. For example:

1. You type `CMD + K`
2. Visual Studio Code waits for the next key of the chord (but doesn't tell you what valid options could be typed)

Instead, what should happen is that VS Code should show the users all of the key chords (or perhaps most commonly used commands *by that specific user*, based on telemetry data), that could possibly be pressed.

This would have the following benefits:

* Significantly improve the user experience
* Ease the learning of new keyboard shortcuts
* Tailor the Visual Studio Code environment to each user (optional)

Cheers,
Trevor Sullivan
Microsoft MVP: PowerShell
https://trevorsullivan.net
https://twitter.com/pcgeek86

**Comment by dakaraphi** [comment]
Eclipse does this and it is very nice.  You have a popup menu after waiting about 1 second.  You can then use the cursor keys to pick an item or finish the key chord.

<img width="211" alt="image" src="https://cloud.githubusercontent.com/assets/10656375/23336204/1cb7fb00-fb97-11e6-8f23-fcf9080497e1.png">

**Comment by jtanner** [comment]
http://spacemacs.org/ has this too and I love it. I makes discovering and learning key bindings easy.

![Spacemacs discoverability](http://spacemacs.org/img/screenshots/ss2.png)

**Comment by CptDarling** [comment]
> Eclipse does this and it is very nice. You have a popup menu after waiting about 1 second. You can then use the cursor keys to pick an item or finish the key chord.
> 
> <img alt="image" width="211" src="https://cloud.githubusercontent.com/assets/10656375/23336204/1cb7fb00-fb97-11e6-8f23-fcf9080497e1.png">

Blender also pops up a menu when there are more chords available.

**Comment by andersennl** [comment]
Has there been any progress on this topic? I came here searching for how to use shortcut chords, thinking VSCode already had this feature.

**Comment by arthurborisow** [comment]
Is there any plans to support it? Or are there any extensions which might help with that? The one above is not exactly what the topic starter meant I guess

**Comment by macintacos** [comment]
Just to expand on this; I think that it's important to have this feature if you want to build up a set of mnemonic chords for shortcuts, similar to how Spacemacs is set up. This, plus the addition of 3-step chords (#6966), would make VSCode really flexible for the users that want to make sure their setup is _exactly_ the way they want it.

**Comment by ashketchupppp** [comment]
+1 to this, I'm trying to move from spacemacs and the which key extension is amazing, I want to setup my keybinds so that all non-text editing keybinds are using chords and it is a pain in the arse. This would make my life so so much easier

**Comment by macintacos** [comment]
Shameless plug, but in this vein I created a keymap called [Chord Heavy](https://marketplace.visualstudio.com/items?itemName=vscode-macintacos.chord-heavy-keymap), inspired by Spacemacs chords. It’d greatly benefit from having this feature added!

**Comment by thiagomajesk** [comment]
I'm also interested in this, I think it would be a great addition. What's the requirement for getting this approved/ voted, etc?

**Comment by DevBodin** [comment]
Was hoping to find it already existed. How frustrating.

**Comment by ashketchupppp** [comment]
Just found this, which may help some folks out. Not at my PC so haven't tried it
https://marketplace.visualstudio.com/items?itemName=VSpaceCode.whichkey&ssr=false#overview

**Comment by ellacrity** [comment]
I think this would greatly enhance the user experience for many, many people. Furthermore, the addition of this feature would, in my humble opinion, help with VSCode product adoption. I use both Neovim and VSCode, but have been transitioning to using the latter much more often lately.

I miss the feature provided by [which-key](https://github.com/folke/which-key.nvim), a popular Neovim plugin that has been mentioned within this ticket above. The plugin emulates the exact behavior that is being requested by this issue. I personally use a lot of key chord combinations, as it greatly increases the number of actions one can do at any given time and makes things (generally) more convenient. Furthermore, VSCode itself has tons of built-in chord combinations.

I truly think it would help others transition from Vim/Neovim to **VSCode** if this feature were to exist, or have better support for it. As a human, it is, unfortunately, very difficult to remember every sequence of keys, all of the time. It's great when we do, but can really interrupt our flow when we forget, and have to refer to our key-bindings again (and again, maybe! ;).

Is there anything that is blocking this issue? I don't see any contributor activity in this issue, so I figured I would ask. I do fully understand that this is mainly a quality-of-life enhancement, and likely has low priority, but it would definitely be a huge enhancement to user/developer experience.

**Comment by macintacos** [comment]
@ellacrity yes, see [VSpaceCode](https://github.com/VSpaceCode/VSpaceCode) and [vscode-which-key](https://github.com/VSpaceCode/vscode-which-key)

**Comment by pawelfras** [comment]
Is there any update on this matter? VSpaceCode](https://github.com/VSpaceCode/VSpaceCode) and [vscode-which-key](https://github.com/VSpaceCode/vscode-which-key) are ok for Vim extension users, but it would be beneficial to have the idea presented by @pcgeek86 built in the VSC, to support native chords.
